src/main/java/dev/morling/onebrc/CalculateAverage_rmannibucau.java

```diff
@@ -0,0 +1,736 @@
+        agg.total++;
+    }
+
+    private int eolNextOffset(final long position, final boolean start) throws IOException {
+        long i = position;
+        while (UNSAFE.getByte(i) != '\n') {
+            i++;
+        }
+        if (UNSAFE.getByte(i) == '\n') {
+            return (int) (i + (start ? 1 : 0) - position);
+        }
+        return (int) (i - position);
+    }
+
+    private static final Unsafe UNSAFE = initUnsafe();
+
+    private static Unsafe initUnsafe() {
+        try {
+            final var theUnsafe = Unsafe.class.getDeclaredField("theUnsafe");
+            theUnsafe.setAccessible(true);
+            return (Unsafe) theUnsafe.get(Unsafe.class);
+        }
+        catch (final NoSuchFieldException | IllegalAccessException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private static class Bytes {
+        private final byte[] value;
+        private final int hash;
+
+        private Bytes(final byte[] value, final int precomputedHash) {
+            this.value = value;
+            this.hash = precomputedHash;
+        }
+
+        @Override
+        public String toString() {
+            return new String(value, UTF_8);
+        }
+
+        @Override
+        public int hashCode() {
+            return hash;
+        }
+
+        @Override
+        public boolean equals(final Object obj) { // we don't need to check the type since it is an internal usage "we know"
+            // important: we used ComputePotentialStationDiffsPerHash to ensure that if the hash is the same there it is the same value
+            // so no need to do anything there
+            return true;
+        }
+    }
+
+    private record Offset(long start, long end) {
+    }
+
+    private record ResultRow(float min, float mean, float max) {
+        public String toString() {
+            return STR."\{round(min)}/\{round(mean)}/\{round(max)}";
+        }
+
+        private double round(final float value) {
+            return Math.round(value * 10.f) / 10.;
+        }
+    }
+
+    private static class MeasurementAggregator {
+        private int min;
+        private int max;
+        private long sum;
+        private long total;
+
+        private MeasurementAggregator(final int min, final int max, final long sum, final long total) {
+            this.min = min;
+            this.max = max;
+            this.sum = sum;
+            this.total = total;
+        }
+    }
+
+    // NOTE: this is a work code which shouldn't stay but kept to explain the rational behind some impl choice (in Bytes) for the exercise
+    // we assume the station set is closed as of today and optimize Bytes equals impl by checking conflicts only per hash
+    public static class ComputePotentialStationDiffsPerHash {
+        private ComputePotentialStationDiffsPerHash() {
+            // no-op
+        }
+
+        public static void main(final String... args) {
+            final var stations = List.of("Abha",
```

Comment:
Hum, there should be no assumptions about the specific data set (see rules). Not quite clear to me what's happening here?

Reply:
well, part of what I'm interesting in there is the extreme abuses of challenges on one side but also the fact that some choices don't work better than default impl (linear hash based structures here).
This main (which is totally useless for the runtime, as written, is really to explain why the impl works) is intended to ensure the hashCode distributes the (pseudo-expected) values properly for a hashmap.
Concretely the idea was to validate "offline" Bytes#equals implementation assuming the names are (pseudo)open but compatible with such an implementation - which is generally the case with such input data.

side note: this was also validated with https://github.com/meteostat/weather-stations (21K+ station names) and even if conflicts are possible they are unlikely enough - or would probably be handled in the file generator (depends a bit the ecosystem around) so for me it is not a so unfair assumption.

That said I can totally understand you think it does not respect the rule, no big deal to not integrate the PR for me.
Don't think I would be in the top results anyway in 20 days with this code and just used holidays time to participate ;).

Reply:
Yeah, indeed I am inclined to close the PR in its current form, as there shouldn't be any specific asumptions or handling of the example station names from the data set, as per the rules. Thanks a lot for participating nevertheless!